ortools/linear_solver/xpress_interface.cc

```diff
@@ -1911,4 +1917,19 @@ void XPRS_CC optimizermsg(XPRSprob prob, void* data, const char *sMsg, int nLen,
 	}
 }
 
+void XpressInterface::SetHint() {
+  // Currently the XPRESS API does not handle clearing out previous hints
+  int const length = solver_->solution_hint_.size();
+  auto* const colind = new int[length];
+  auto* const solval = new double[length];
```

Comment:
This looks like a memory leak : does XPRESS own this memory and deallocate it at some point ?

Even if XPRESS did deallocate memory (I doubt it), being a C library, it would call `free`. Mixing `new` and `free` is UB.

If you're unsure, you can call Valgrind on the test you wrote to check for memory leaks.